client/src/lib/actions/profile_setup.ts

```diff
@@ -21,63 +34,171 @@ export type ProfileSetupState = {
   message?: string | null;
 };
 
-export async function createProfile(
-  prevState: ProfileSetupState,
-  formData: FormData
-): Promise<ProfileSetupState> {
-  console.log("createProfile");
-
-  // Validate form data with Zod schema
-  const validatedFields = ProfileSetupFormSchema.safeParse({
-    name: formData.get("name"),
-    username: formData.get("username"),
-    school: formData.get("school"),
-    avatar: formData.get("avatar"),
-  });
+export async function userHasExistingProfile() : Promise<boolean>{
+  debug("start of userHasExistingProfile() function");
+  const userProfile = await getProfileForUser();
+  return (userProfile) ? true: false;
+} // end of userHasExistingProfile
 
-  if (!validatedFields.success) {
-    console.log("validation failed");
-    return {
-      errors: validatedFields.error.flatten().fieldErrors,
-      message: "Incorrect fields. Create Profile Failed",
-    };
+export async function getUser(): Promise<Json> {
+  debug("start of getUser() function");
+  const supabase = createSupabaseServer();
+  const { data: {user}, error: userError } =  await supabase.auth.getUser();
+  if (userError) {
+      return { 
+        errors: {
+          school: ["Error: No user"],
+        }
+      }
   }
+  const jsonDataStr = JSON.stringify(user, null, 2);
+  const jsonData = JSON.parse(jsonDataStr);
+  return jsonData;
+} // end of getUser
 
-  const { username, school, avatar } = validatedFields.data;
+export async function getUserId() : Promise<string|null> {
```

Comment:
This function is redundant with the getUser function. Why not just return the ID of the user in getUser() function?